### DISCUSSION TITLE
Netbox on RedHat Linux

### DISCUSSION THREAD

**Discussion by berkeleycowboy** [netbox-community/netbox#13394] [General]
I've been attempting to build a Dev Netbox for the past several months.  

The instructions for the PostgreSQL installation indicates the following command should be used:

     sudo yum install -y postgresql-server

This command does not install the required PostgreSQL version on Red Hat Enterprise Linux vers 8.8.  PostgresSQL version 10 is installed.  

This site also uses the same command but when the version is tested its only version 10.17
https://www.centlinux.com/2021/11/install-netbox-on-centos-red-hat.html

I'm wondering if the PostgresSQL installation is out of date for CentOS \ RedHat ?

The install-netbox-on-centos-red-hat.html URL also makes use of a "Soft Link" which the official Netbox write up does not utilize.  

I have performed the installation  several times but have never been able to obtain a working configuration.  This is why I have started comparing different configuration installation to see if there's anything that 's missing.  

Thanks in advance for any help that can be provided.

**Comment by mtinberg** [comment]
I've been successfully using PG12 from the yum/dnf repo from postgresql.com rather than the built-in default version or one from EPEL or SCL. Those are the repos I'd look for a supportable package, built-in, SCL, EPEL and vendor, in that order.  I don't know if there is a last-updated-time or last-updated-netbox-version on the docs to hint when they may be out-of-date because the software requirements or the platform have changed since the procedure was last tested, but things do change and when the docs are no longer copy-paste accurate they still can provide a great guideline as to what the intended outcome of each step is, even if you need to be flexible on how those steps are accomplished.

Some of this confusion may be from not understanding Redhat software management policies, they support the version of software that shipped with an EL release for the entire lifetime of that OS, with a few notable exceptions for fast-changing desktop software, so if EL8 shipped with PG10 then it will only contain PG10 until the end of time, with patches added onto that PG10 throughout that lifetime.  This is great for running "enterprise" applications over long periods of time by keeping the OS stable so as to not incur unexpected maintenance costs, but more agile applications can change the needed versions of dependencies over that time span as they are often developed against the latest release of their libraries, not a fixed version.

So Redhat provides Software Collections (SCL) which packages newer versions of common dependencies (like Python) as separately installable packages, which is what I'm using on my EL7 hosts to run Netbox in rh-python38 and not the system python36 which is no longer supported.  Newer versions of PostgreSQL might be available from those repos as well, if not then the PostgreSQL project provides repos with every supported (by them) release built against several Enterprise Linux versions, so you can run the latest if that is what your app needs.

If you work through the inconsistencies, you can provide some diffs/patches/pull requests for the documentation so the next person who comes along is better off.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Dennis Weikel ***@***.***>
Sent: Monday, August 7, 2023 12:53 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Netbox on RedHat Linux (Discussion #13394)


I've been attempting to build a Dev Netbox for the past several months.

The instructions for the PostgreSQL installation indicates the following command should be used:

 sudo yum install -y postgresql-server


This command does not install the required PostgreSQL version on Red Hat Enterprise Linux vers 8.8. PostgresSQL version 10 is installed.

This site also uses the same command but when the version is tested its only version 10.17
https://www.centlinux.com/2021/11/install-netbox-on-centos-red-hat.html

I'm wondering if the PostgresSQL installation is out of date for CentOS \ RedHat ?

The install-netbox-on-centos-red-hat.html URL also makes use of a "Soft Link" which the official Netbox write up does not utilize.

I have performed the installation several times but have never been able to obtain a working configuration. This is why I have started comparing different configuration installation to see if there's anything that 's missing.

Thanks in advance for any help that can be provided.

**Comment by Snarf42** [comment]
RHEL 8.8 supports the following postgreSQL versions: 9.6, 10, 12, 13, and 15

generic link to documentation:
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8

you can check which modules are supported with
`yum module list postgresql`

more information https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/installing_managing_and_removing_user-space_components/managing-versions-of-appstream-content_using-appstream

**Comment by berkeleycowboy** [comment]
I've successfully installed postgreSQL version 13.  

The upgrade script appears to have run successfully.  I see lots of references to django being installed.  

Super User Created.  
Housekeeping Task Scheduled

When testing  the Application I receive the following output.  

[dennis@scdl-ifipam-001 netbox]$ python3 manage.py runserver 0.0.0.0:8000 --insecure
Traceback (most recent call last):
  File "/opt/netbox-3.5.8/netbox/manage.py", line 8, in <module>
    from django.core.management import execute_from_command_line
ModuleNotFoundError: No module named 'django'
[dennis@scdl-ifipam-001 netbox]$

Manage.py appears to fail at " from django.core.management import execute_from_command_line"

#!/usr/bin/env python
import os
import sys

if __name__ == "__main__":
    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "netbox.settings")

    from django.core.management import execute_from_command_line

    execute_from_command_line(sys.argv)


Not sure what could be occurring or how to troubleshoot at this point.

**Reply by candlerb** [reply]
You need to run manage.py while the virtualenv is activated.  Or, replace `python3` with `/opt/netbox/venv/bin/python`

**Comment by berkeleycowboy** [comment]
I was able finally able to validate the Netbox installation.  

(venv) [root@scdl-ifipam-001 netbox]# python3 manage.py runserver 0.0.0.0:8000 --insecure
Performing system checks...

System check identified no issues (0 silenced).
August 17, 2023 - 19:10:36
Django version 4.1.10, using settings 'netbox.settings'
Starting development server at http://0.0.0.0:8000/

Assuming the server is utilizing 0.0.0.0 instead 127.0.0.0 because of the following configuration in the configuration.py
ALLOWED_HOSTS = ['*']

**Reply by candlerb** [reply]
"0.0.0.0" means "listen on all IPv4 interface addresses"
